src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,134 @@
+                    entity.send(affectBodyTemperatureEvent);
+                    deltaTemp = affectBodyTemperatureEvent.getResultValue();
+                } else {
+                    AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(-1 * deltaTemp);
+                    entity.send(affectBodyTemperatureEvent);
+                    deltaTemp = -1 * affectBodyTemperatureEvent.getResultValue();
+                }
+                //Check for change in body temperature levels.
+                BodyTemperatureLevel before = lookupLevel(bodyTemperature.current);
+                float oldValue = bodyTemperature.current;
+                //Update current body temperature.
+                bodyTemperature.current = bodyTemperature.current + deltaTemp;
+                entity.saveComponent(bodyTemperature);
+                float newValue = bodyTemperature.current;
+                BodyTemperatureLevel after = lookupLevel(bodyTemperature.current);
+
+                if (before != after) {
+                    entity.send(new BodyTemperatureLevelChangedEvent(before, after));
+                    bodyTemperature.currentLevel = after;
+                }
+
+                if(oldValue != newValue) {
+                    entity.send(new BodyTemperatureValueChangedEvent(oldValue, newValue));
+                }
+
+                //only for development purposes
+                entity.getOwner().send(new ChatMessageEvent("Body Temperature: " + bodyTemperature.current,
+                        entity.getOwner()));
+                entity.getOwner().send(new ChatMessageEvent("Env Temperature: " + envTemperature, entity.getOwner()));
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureChangedToHigh(BodyTemperatureLevelChangedEvent event, EntityRef player) {
+        if (event.getNewBodyTemperatureLevel() == BodyTemperatureLevel.HIGH) {
+            player.addOrSaveComponent(new HyperthermiaComponent());
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureChangedToLow(BodyTemperatureLevelChangedEvent event, EntityRef player) {
+        if (event.getNewBodyTemperatureLevel() == BodyTemperatureLevel.LOW) {
+            player.addOrSaveComponent(new HypothermiaComponent());
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureChangedToNormal(BodyTemperatureLevelChangedEvent event, EntityRef player) {
+        if (event.getNewBodyTemperatureLevel() == BodyTemperatureLevel.NORMAL) {
+            if (player.hasComponent(HyperthermiaComponent.class)) {
+                player.removeComponent(HyperthermiaComponent.class);
+            }
+            if (player.hasComponent(HypothermiaComponent.class)) {
+                player.removeComponent(HypothermiaComponent.class);
+            }
+        }
+    }
+
+    public BodyTemperatureLevel lookupLevel(float temperature) {
```

Comment:
Guess this is more of a mapping than a lookup. Lookup sounds like you check what the current body temperature level of a player is instead of checking which body temperature level a temperature maps to.